GVFS/FastFetch/NativeMethods.cs

```diff
@@ -0,0 +1,35 @@
+﻿using GVFS.Common.Tracing;
+using System.IO.MemoryMappedFiles;
+using System.Runtime.InteropServices;
+
+namespace FastFetch
+{
+    internal static class NativeMethods
+    {
+        public static bool isUnixOS = RuntimeInformation.IsOSPlatform(OSPlatform.OSX) || RuntimeInformation.IsOSPlatform(OSPlatform.Linux);
+
+        public static unsafe void WriteFile(ITracer tracer, byte* originalData, long originalSize, string destination, ushort mode)
+        {
+            if (!isUnixOS)

```

Comment:
Super minor style comment (and this might just be me), but I think this code (and below in `StatAndUpdateIndexForFile`) would be a bit more readable if you preferred affirmative checks (i.e. `if (isUnixOS)`) rather than negative checks (`if (!isUnixOS)`).